src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/ClientVersionLimitRule.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Consensus.Rules;
+
+namespace Stratis.Bitcoin.Features.Consensus.Rules.CommonRules
+{
+    /// <summary>Checks if <see cref="Block"/> the client is supported by the network.</summary>
+    public class ClientVersionLimitRule : HeaderValidationConsensusRule
+    {
+        /// <inheritdoc />
+        /// <exception cref="ConsensusErrors.ClientVersionTooOld">Thrown if client version is too old.</exception>
+        public override void Run(RuleContext context)
+        {
+            ChainedHeader chainedHeader = context.ValidationContext.ChainedHeaderToValidate;
+            int maxSupportedSyncBlockHeight = this.Parent.Network.Consensus.Options.MaxSupportedSyncBlockHeight;
+            int maxSupportedBlockHeightGracePeriod = this.Parent.Network.Consensus.Options.MaxSupportedBlockHeightGracePeriod;
+
+            // Check if the client version is approaching end of life
+            if ((maxSupportedBlockHeightGracePeriod > 0) && (chainedHeader.Height > (maxSupportedSyncBlockHeight - maxSupportedBlockHeightGracePeriod)))
```

Comment:
Should we also add `&& (chainedHeader.Height <= maxSupportedSyncBlockHeight)` so we don't show the warning when we are past the max height and techically we "reached end life" and error will be thrown by the next `if` block